### DISCUSSION TITLE
Packaging error on a non java plugin (Tycho 2.7.5)

### DISCUSSION THREAD

**Discussion by wimjongman** [eclipse-tycho/tycho#1372] [Q&A] [answered]
I have a bundle that causes Tycho 2.7.5 to report an error. 

This bundle did not have an issue before. The weird thing about this is that it is a plugin project, but not a java project. It is used to carry licenses and stuff, not java code.

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:58 min
[INFO] Finished at: 2022-09-20T13:35:28+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:2.7.5:package-feature (default-package-feature) on project org.eclipse.birt.chart: Execution default-package-feature of goal org.eclipse.tycho:tycho-packaging-plugin:2.7.5:package-feature failed: Duplicate key org.eclipse.swt (attempted merging values org.eclipse.tycho.model.Feature$ImportRef@4a0ebdce and org.eclipse.tycho.model.Feature$ImportRef@3902dc9d) -> [Help 1]

**Comment by laeubi** [comment, marked as answer]
Can you link the feature? This usually indicates that you have the same (competing) entry for a dependency in your feature (according to the error it is `org.eclipse.swt` that is mentioned multiple times in your feature.

> This bundle did not have an issue before.
>  The weird thing about this is that it is a plugin project, but not a java project.

According to the message, it is treated as a feature, is it possible that you are using pomless and your project contains a bundle + a feature.xml?

**Reply by wimjongman** [reply]
I have a project org.eclipse.birt.chart[1] but also a maven group [2]. It is here[2] where it reports the issue. I thought it was at [1] (which is the non-java plugin project)

[1] https://github.com/eclipse/birt/tree/master/build
[2] https://github.com/eclipse/birt/tree/master/chart

**Reply by wimjongman** [reply]
I have looked at the chart feature, and indeed it contains a double SWT entry. I have fixed this, and I am building again.

**Reply by wimjongman** [reply]
Yes, that was the issue. The error is a bit confusing because it is on the group and not on the actual feature.

**Comment by apupier** [comment]
Hello,

I guess it is because you upgraded Tycho to 2.7.5 which is reporting
duplicated bundles in a feature now.
it seems that org.eclipse.swt is defined two times in the feature:
https://github.com/eclipse/birt/blob/43e8bf06ddaaaf9b97f2eeeb93df21a1456096ea/features/org.eclipse.birt.chart.feature/feature.xml#L43
https://github.com/eclipse/birt/blob/43e8bf06ddaaaf9b97f2eeeb93df21a1456096ea/features/org.eclipse.birt.chart.feature/feature.xml#L45

Removing one th entry should do the trick.
